### DISCUSSION TITLE
gulp-inject configuration

### DISCUSSION THREAD

**Discussion by pschaeflein** [pnp/generator-teams#276] [Ideas]
I would like to propose an update to support a developer-provided configuration for gulp-inject.

The yoteams-build-core package [supports configuration](https://github.com/pnp/generator-teams/tree/master/packages/yoteams-build-core#configuration) to extend what file paths are processed, but does not provide for changing the processing.

I would appreciate thoughts around extending the config object to include injectOptions:

```typescript
interface IYoTeamsBuildCoreConfig {
  watches: string[];
  clientWatches: string[];
  staticFiles: string[];
  injectSources: string[];
  injectOptions: any;  // @types/gulp-inject: IOptions
  htmlFiles: string[];
}
```

(Options for gulp-inject: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e605162d378f52f5cb8fccb137442320c876bf4d/types/gulp-inject/index.d.ts#L22)

The specified injectOptions (if any) would then be used in the webTasks.injectSources method instead of the default.

**Comment by wictorwilen** [comment]
This is a great idea @pschaeflein!

**Reply by pschaeflein** [reply]
PR incoming this week. ;)

**Reply by pschaeflein** [reply]
I created an issue, but cannot assign it. Pls assign to me.

**Comment by wictorwilen** [comment]
Feature implemented
